runtime/src/accounts_db.rs

```diff
@@ -1029,13 +1058,13 @@ impl AccountsDB {
                     let sparse_by_count = (alive_count as f32 / stored_count as f32) <= 0.8;
                     let sparse_by_bytes = (written_bytes as f32 / total_bytes as f32) <= 0.8;
```

Comment:
I think we're better off skipping to shrink if total_bytes are around PAGE_SIZE or 2*PAGE_SIZE. Otherwise, we'll start to shrink small appendvecs too frequently for almost no win after this pr (yeah, I could have done at #14072):

(This is taken from mainnet-beta while shrinking old roots)

```
[2020-12-21T16:06:08.052216524Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56363861): skip_shrink: false count: 3/6 byte: 1485/4096
[2020-12-21T16:06:08.310957737Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56480921): skip_shrink: false count: 5/6 byte: 1821/4096
[2020-12-21T16:06:08.531950417Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56456146): skip_shrink: false count: 25/28 byte: 8341/12288
[2020-12-21T16:06:08.738150058Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56351547): skip_shrink: false count: 7/8 byte: 2653/4096
[2020-12-21T16:06:09.746224464Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56199914): skip_shrink: false count: 7/8 byte: 2261/4096
[2020-12-21T16:06:09.846730032Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56431829): skip_shrink: true count: 34/37 byte: 10573/12288
[2020-12-21T16:06:10.175186883Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56487717): skip_shrink: true count: 44/48 byte: 14085/16384
[2020-12-21T16:06:10.633622902Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56388477): skip_shrink: false count: 7/8 byte: 2429/4096
[2020-12-21T16:06:10.950872969Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56222118): skip_shrink: false count: 4/5 byte: 1517/4096
[2020-12-21T16:06:11.505014070Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56454911): skip_shrink: true count: 11/13 byte: 3781/4096
[2020-12-21T16:06:11.754149803Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56535189): skip_shrink: false count: 16/17 byte: 5165/8192
[2020-12-21T16:06:11.959589737Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56188165): skip_shrink: false count: 2/3 byte: 909/4096
[2020-12-21T16:06:12.376137359Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56435798): skip_shrink: false count: 15/16 byte: 4693/8192
[2020-12-21T16:06:12.714198710Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56334004): skip_shrink: false count: 4/5 byte: 1013/4096
[2020-12-21T16:06:13.515383761Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56465584): skip_shrink: false count: 9/10 byte: 2869/4096
[2020-12-21T16:06:15.517786062Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56350686): skip_shrink: false count: 6/9 byte: 2397/4096
[2020-12-21T16:06:16.418928396Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56552396): skip_shrink: true count: 59/60 byte: 21069/24576
[2020-12-21T16:06:17.119784717Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56361031): skip_shrink: false count: 6/7 byte: 1957/4096
[2020-12-21T16:06:17.220099106Z INFO  solana_runtime::accounts_db] shrink_stale_slot (56428561): skip_shrink: false count: 5/6 byte: 1821/4096
```

Reply:
Yea, I did experiment with this as well. The effect is not large, but I can add that change.